components/arm/universalrobots/ur.go

```diff
@@ -471,30 +471,29 @@ func reader(ctx context.Context, conn io.Reader, ua *URArm, onHaveData func()) e
 	}
 }
 
-const errBuiltinKinematics = "cannot use UR builtin kinematics with obstacles or interaction spaces"
+const errURHostedKinematics = "cannot use UR hosted kinematics with obstacles or interaction spaces"
 
-func (ua *URArm) useBuiltinKinematics(worldState *commonpb.WorldState, extra map[string]interface{}) (bool, error) {
-	// function to error out if trying to use world state with builtin kinematics
-	checkWorldState := func(usingBuiltinKinematics bool) (bool, error) {
-		if usingBuiltinKinematics && worldState != nil && (len(worldState.Obstacles) != 0 || len(worldState.InteractionSpaces) != 0) {
-			return false, errors.New(errBuiltinKinematics)
+func (ua *URArm) useURHostedKinematics(worldState *commonpb.WorldState, extra map[string]interface{}) (bool, error) {
+	// function to error out if trying to use world state with hosted kinematics
+	checkWorldState := func(usingHostedKinematics bool) (bool, error) {
+		if usingHostedKinematics && worldState != nil && (len(worldState.Obstacles) != 0 || len(worldState.InteractionSpaces) != 0) {
+			return false, errors.New(errURHostedKinematics)
 		}
-		return usingBuiltinKinematics, nil
+		return usingHostedKinematics, nil
 	}
 
 	// if runtime preference is specified, obey that
 	if extra != nil {
-		if usingAtRuntime, ok := extra["builtin_kinematics"].(bool); ok {
+		if usingAtRuntime, ok := extra["ur_hosted_kinematics"].(bool); ok {
```

Comment:
should we make this valid for other arms too?
"arm_hosted_kinematics?

Reply:
I like this.  Do you see an issue in generalizing now if we don't support it in any of the other drivers yet?

Reply:
no

Reply:
Changed